### PR TITLE
Optimise ImageOps.fit by combining resize and crop

### PR DESCRIPTION
**Pull request by homm**
This optimises crop+resize operations sequence by using `box` parameter of `resize` function. This gives us:

1. Performance. Results for [lilliput-bench](
https://github.com/discordapp/lilliput-bench):

Before:

```
JPEG 256x256 => 32x32:  1109 Bytes,     avg: 1.43 ms    min: 0.92 ms    max: 4.43 ms
PNG 256x256 => 32x32:   2603 Bytes,     avg: 3.70 ms    min: 3.62 ms    max: 7.14 ms
WEBP 256x256 => 32x32:  654 Bytes,      avg: 2.77 ms    min: 2.48 ms    max: 6.30 ms
GIF 256x256 => 32x32:   1897 Bytes,     avg: 4.89 ms    min: 4.51 ms    max: 11.58 ms
JPEG 1920x1080 => 800x600:      107480 Bytes,   avg: 26.28 ms   min: 24.90 ms   max: 37.08 ms
PNG 1920x1080 => 800x600:       810489 Bytes,   avg: 282.60 ms  min: 277.43 ms  max: 377.78 ms
WEBP 1920x1080 => 800x600:      84960 Bytes,    avg: 126.52 ms  min: 123.86 ms  max: 149.28 ms
GIF 1920x1080 => 800x600:       411470 Bytes,   avg: 102.40 ms  min: 95.52 ms   max: 126.27 ms
```

After:

```
JPEG 256x256 => 32x32:	1109 Bytes,	avg: 0.95 ms	min: 0.74 ms	max: 4.93 ms
PNG 256x256 => 32x32:	2603 Bytes,	avg: 3.56 ms	min: 3.45 ms	max: 7.60 ms
WEBP 256x256 => 32x32:	654 Bytes,	avg: 2.77 ms	min: 2.47 ms	max: 6.36 ms
GIF 256x256 => 32x32:	1897 Bytes,	avg: 4.73 ms	min: 4.37 ms	max: 11.39 ms
JPEG 1920x1080 => 800x600:	107450 Bytes,	avg: 21.36 ms	min: 20.99 ms	max: 27.02 ms
PNG 1920x1080 => 800x600:	810516 Bytes,	avg: 277.31 ms	min: 272.55 ms	max: 328.37 ms
WEBP 1920x1080 => 800x600:	84952 Bytes,	avg: 125.57 ms	min: 121.98 ms	max: 154.98 ms
GIF 1920x1080 => 800x600:	411470 Bytes,	avg: 98.35 ms	min: 94.72 ms	max: 117.11 ms
```

2. Accuracy. Unlike crop method, `box` for resize method accepts float coordinates. For example:

```python
from PIL import Image, ImageOps
dot = Image.open('../dot.png')
ImageOps.fit(dot, (220, 200), Image.BICUBIC).save('../_out.png')
```

Source image (50x30px), processed image before, processed image after:

screenshot 2018-10-17 at 02 30 15
screenshot 2018-10-17 at 02 30 04
screenshot 2018-10-17 at 02 31 40

### CODE REVIEW

File: src/PIL/ImageOps.py [hunk 1]
```diff
@@ -400,66 +401,50 @@ def fit(image, size, method=Image.NEAREST, bleed=0.0, centering=(0.5, 0.5)):
...
-    if bleed > 0.0:
-        liveArea = (
-            bleedPixels[0], bleedPixels[1], image.size[0] - bleedPixels[0] - 1,
-            image.size[1] - bleedPixels[1] - 1
-            )
+    bleed_pixels = (bleed * image.size[0], bleed * image.size[1])
 
-    liveSize = (liveArea[2] - liveArea[0], liveArea[3] - liveArea[1])
+    live_size = (image.size[0] - bleed_pixels[0] * 2,
+                image.size[1] - bleed_pixels[1] * 2)
```
Comment (hugovk) [inline src/PIL/ImageOps.py]: Add an extra space to align `image.size` in these two lines

**Comment by hugovk** [comment]
Let's remember to add this to the release notes.

**Comment by homm** [comment]
@hugovk Added release notes. Could you check?

File: docs/releasenotes/5.4.0.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+-----
+
+
+Other Changes
+=============
+
+ImageOps.fit
+^^^^^^^^^^^^
+
+Now uses the one resize operation with ``box`` argument internally
```
Comment (hugovk) [inline docs/releasenotes/5.4.0.rst]:
```suggestion
Now uses one resize operation with ``box`` argument internally
```

File: docs/releasenotes/5.4.0.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+instead of the crop and scale operations sequence.
```
Comment (hugovk) [inline docs/releasenotes/5.4.0.rst]:
```suggestion
instead of a crop and scale sequence.
```

File: docs/releasenotes/5.4.0.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+This improves the performance and accuracy of cropping since
+``box`` parameter accepts float values.
```
Comment (hugovk) [inline docs/releasenotes/5.4.0.rst]:
```suggestion
the ``box`` parameter accepts float values.
```

**Pull request merged**